manifests/init.pp

```diff
@@ -17,10 +17,12 @@
 #   TODO this seems like something we should derive.
 #
 class java_artisanal (
-  Stdlib::HTTPUrl $source = 'https://github.com/frekele/oracle-java/releases/download/8u202-b08/jdk-8u202-linux-x64.rpm',
+  Stdlib::HTTPUrl $source  = 'https://github.com/frekele/oracle-java/releases/download/8u202-b08/jdk-8u202-linux-x64.rpm',
   String          $package = 'jdk1.8',
   String          $dir     = 'jdk1.8.0_202-amd64',
 ) {
+  include java_artisanal::java17
```

Comment:
Coupling the classes together means that they will practically be usable outside of the CCS context.  That is probably acceptable as there are not currently any other users.  My preference would be completely decouple them, as it allows for more flexible composition at the role/profile level, and to have CCS related roles include both classes.  I concede this is fairly subject design pattern preferences on my part.  How do you feel about a completely separation?